x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ssl/PemUtils.java

```diff
@@ -176,7 +177,7 @@ private static PrivateKey parsePKCS8(BufferedReader bReader) throws IOException,
             line = bReader.readLine();
         }
         if (null == line || PKCS8_FOOTER.equals(line.trim()) == false) {
-            throw new IOException("Malformed PEM file, PEM footer is invalid or missing");
+            throw new KeyException("Malformed PEM file, PEM footer is invalid or missing");
```

Comment:
This change was needed because `readPrivateKey` no longer catches & wraps IOException (which, in turn, was needed in order to have specific handling of `NoSuchFileException`)